### PR TITLE
refactor: migrate moment to date-fns

### LINKED ISSUE #140: [suggestion] migration moment.js to date-fns
**Description**:
migration moment.js to date-fns

**Why**:
Moment is lagacy project. And it might cause a huge performance overhead because of its complex APIs and large bundle size.

[moment docs](https://momentjs.com/docs)
[You-Dont-Need-Momentjs](https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md)

### PR DESCRIPTION
**Pull request by DongjaJ**
#### What this PR does / why we need it?
Moment is a lagacy project. And it might cause a huge performance overhead because of its complex APIs and large bundle size. So this PR migrations moment.js to date-fns

#### What are the relevant tickets?

Fixes #140 

### Checklist
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: src/features/documents/DocumentList.tsx [hunk 1]
```diff
@@ -210,7 +210,7 @@ export function DocumentList({ isDetailOpen = false }: { isDetailOpen?: boolean
                     <span className="td id">{key}</span>
                     {!isDetailOpen && (
                       <span className="td updated">
-                        {moment.unix(updatedAt).format(`${use24HourClock ? 'MMM D, H:mm' : 'MMM D, h:mm A'}`)}
+                        {format(fromUnixTime(updatedAt), `${use24HourClock ? 'MMM d, h:mm' : 'MMM d, h:mm A'}`)}
```
Comment (hackerwins) [inline src/features/documents/DocumentList.tsx:213]:
```suggestion
                        {format(fromUnixTime(updatedAt), `${use24HourClock ? 'MMM d, h:mm' : 'MMM d, h:mm aa'}`)}
```

`A` should be changed to `aa` according to the docs.

https://date-fns.org/v3.6.0/docs/format
Screenshot 2024-04-15 at 8 47 52 PM

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thank you for introducing date-fns.
I left a few comments.

File: src/features/documents/DocumentList.tsx (hunk 1 above)
Comment (DongjaJ) [reply src/features/documents/DocumentList.tsx]: Thanks for your review! I changed it

**Comment by hackerwins** [comment]
Thanks for your contribution.
This was merged by https://github.com/yorkie-team/dashboard/commit/caae3547329c8a73c2718d75d5218d1d7a978a74.

**Pull request closed without merging**